backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -340,4 +355,81 @@ private static Predicate<? super SonarLintRuleDefinition> isIncludedByConfigurat
       return false;
     };
   }
+
+  @EventListener
+  public void onPluginsSynchronized(PluginsSynchronizedEvent event) {
+    checkIfReadyForAnalysis(configurationRepository.getBoundScopesToConnection(event.getConnectionId())
+      .stream().map(BoundScope::getConfigScopeId).collect(Collectors.toSet()));
+  }
+
+  @EventListener
+  public void onConfigurationScopeAdded(ConfigurationScopesAddedEvent event) {
+    checkIfReadyForAnalysis(event.getAddedConfigurationScopeIds());
+  }
+
+  @EventListener
+  public void onConfigurationScopeRemoved(ConfigurationScopeRemovedEvent event) {
+    analysisReadinessByConfigScopeId.remove(event.getRemovedConfigurationScopeId());
+  }
+
+  @EventListener
+  public void onBindingConfigurationChanged(BindingConfigChangedEvent event) {
+    var configScopeId = event.getConfigScopeId();
+    analysisReadinessByConfigScopeId.remove(configScopeId);
+    if (!checkIfReadyForAnalysis(configScopeId)) {
+      client.didChangeAnalysisReadiness(new DidChangeAnalysisReadinessParams(Set.of(configScopeId), false));
```

Comment:
Shouldn't the sending of the event managed at the same place? In the `saveAndNotifyReadyForAnalysis`?

Reply:
It could, but the method would be a bit more complex